backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -551,4 +560,36 @@ private static void checkSystemEndpoint(EndpointType endpointType) {
     private boolean isEndpointTypeAllowed(EndpointType endpointType) {
         return !featureFlipper.isEmailsOnlyMode() || endpointType.isSystemEndpointType;
     }
+
+    /**
+     * Removes the secrets from the endpoint's properties when returning them
+     * to the client.
+     * @param endpoint the endpoint to redact the secrets from.
+     */
+    @Deprecated(forRemoval = true)
+    protected void redactSecretsForEndpoint(final SecurityContext securityContext, final Endpoint endpoint) {
+        // Only redact the secrets for those users who have read only permissions.
+        if (securityContext.isUserInRole(ConsoleIdentityProvider.RBAC_READ_INTEGRATIONS_ENDPOINTS)
+            || securityContext.isUserInRole(ConsoleIdentityProvider.RBAC_READ_NOTIFICATIONS)
+            || securityContext.isUserInRole(ConsoleIdentityProvider.RBAC_READ_NOTIFICATIONS_EVENTS)
+        ) {
```

Comment:
Shouldn't this be
```suggestion
        if (!securityContext.isUserInRole(ConsoleIdentityProvider.RBAC_WRITE_INTEGRATIONS_ENDPOINTS)) {
```
?